packages/contracts/contracts/generator/GenArt721GeneratorV0.sol

```diff
@@ -0,0 +1,331 @@
+                .projectScriptDetails(projectId)
+        returns (string memory, string memory, uint256 scriptCount) {
+            if (scriptCount == 0) {
+                return "";
+            }
+
+            address[] memory scriptBytecodeAddresses = new address[](
+                scriptCount
+            );
+
+            for (uint256 i = 0; i < scriptCount; i++) {
+                scriptBytecodeAddresses[i] = IGenArt721CoreProjectScriptV1(
+                    coreContractAddress
+                ).projectScriptBytecodeAddressByIndex(projectId, i);
+            }
+
+            bytes32 storageVersion = BytecodeStorageReader
+                .getLibraryVersionForBytecode(scriptBytecodeAddresses[0]);
+
+            uint256 offset;
+            if (storageVersion == BytecodeStorageReader.V0_VERSION_STRING) {
+                offset = 104;
+            } else if (
+                storageVersion == BytecodeStorageReader.V1_VERSION_STRING
+            ) {
+                offset = 65;
+            } else {
+                revert("Unsupported storage version");
+            }
+
+            return AddressChunks.mergeChunks(scriptBytecodeAddresses, offset);
+        } catch {
+            // Noop try again for older contracts.
+        }
+
+        try
+            IGenArt721CoreProjectScriptV0(coreContractAddress)
+                .projectScriptInfo(projectId)
+        returns (string memory, uint256 scriptCount) {
+            if (scriptCount == 0) {
+                return "";
+            }
+
+            string memory script;
+            for (uint256 i = 0; i < scriptCount; i++) {
+                string memory scriptChunk = IGenArt721CoreProjectScriptV0(
+                    coreContractAddress
+                ).projectScriptByIndex(projectId, i);
+                script = string.concat(script, scriptChunk);
+            }
+
+            return abi.encodePacked(script);
+        } catch {
+            revert("Unable to retrieve project script info");
+        }
+    }
+
+    /**
+     * @notice Get the HTMLRequest for a given token.
+     * @param coreContractAddress The core contract address the token belongs to.
+     * @param tokenId The ID of the token to retrieve the HTMLRequest for.
+     * @return The HTMLRequest for the token.
+     */
+    function getTokenHtmlRequest(
+        address coreContractAddress,
+        uint256 tokenId
+    ) internal view returns (HTMLRequest memory) {
+        _onlySupportedCoreContract(coreContractAddress);
+
+        uint256 projectId = tokenId / ONE_MILLION;
+        // This will revert for older contracts that do not have an override set.
+        bytes32 dependencyNameAndVersion = dependencyRegistry
+            .getDependencyNameAndVersionForProject(
+                coreContractAddress,
+                projectId
+            )
+            .stringToBytes32();
+
+        bytes32 tokenHash;
+        if (tokenHash == bytes32(0)) {
+            try
```

Comment:
nit: suggest adding some dev comments about which core versions are being tried along the way here